src/Adapter/Product/Repository/ProductRepository.php

```diff
@@ -64,6 +84,57 @@ public function assertProductExists(ProductId $productId): void
         $this->assertObjectModelExists($productId->getValue(), 'product', ProductNotFoundException::class);
     }
 
+    /**
+     * @param ProductId[] $productIds
+     *
+     * @throws ProductNotFoundException
+     */
+    public function assertProductsExists(array $productIds): void
```

Comment:
I wonder if this could be implemented in the Abstract class in some ways I guess many repositories might need this methods to check if products exists as well
Although this could be done simply by looping through the array and using the existing `assertExists`

Reply:
Also maybe we could find a less confusing function name, it's only different by a letter `s` so it can easily been mixed up using auto completion (and the pending s is actually a conjugation mistake)
Maybe we could use something like `assertProductListExists` `assertProductCollectionExists` or `assertAllProductsExist`